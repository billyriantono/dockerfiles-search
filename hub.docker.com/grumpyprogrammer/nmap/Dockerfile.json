{
  "MetaArgs": null,
  "Stages": [
    {
      "Name": "",
      "BaseName": "alpine:3.10.3",
      "SourceCode": "FROM alpine:3.10.3",
      "Platform": "",
      "From": {
        "Image": "alpine:3.10.3"
      },
      "Commands": [
        {
          "Key": "NMAP_VERSION",
          "Name": "arg",
          "Value": "7.70-r4"
        },
        {
          "CmdLine": [
            "apk add --update --no-cache     nmap=${NMAP_VERSION}"
          ],
          "Name": "run",
          "PrependShell": true
        }
      ]
    }
  ]
}