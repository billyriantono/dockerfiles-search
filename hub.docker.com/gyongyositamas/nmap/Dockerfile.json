{
  "MetaArgs": null,
  "Stages": [
    {
      "Name": "",
      "BaseName": "ubuntu",
      "SourceCode": "FROM ubuntu",
      "Platform": "",
      "From": {
        "Image": "ubuntu"
      },
      "Commands": [
        {
          "CmdLine": [
            "apt-get update"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "apt-get -y install mc"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "apt-get -y install nmap"
          ],
          "Name": "run",
          "PrependShell": true
        }
      ]
    }
  ]
}