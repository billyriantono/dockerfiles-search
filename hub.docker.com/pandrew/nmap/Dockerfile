FROM alpine:edge

ENV	NMAP_VERSION 7.40-r0
RUN	apk --no-cache add \
		nmap=$NMAP_VERSION \
		nmap-scripts=$NMAP_VERSION \
		nmap-nselibs=$NMAP_VERSION

ENTRYPOINT ["nmap"]
