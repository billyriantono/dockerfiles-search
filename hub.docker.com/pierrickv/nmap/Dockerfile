FROM centos:7

RUN rpm -vhU https://nmap.org/dist/nmap-7.70-1.x86_64.rpm

ENTRYPOINT ["nmap"]
