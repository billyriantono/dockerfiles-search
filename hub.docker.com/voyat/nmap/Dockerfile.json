{
  "MetaArgs": null,
  "Stages": [
    {
      "Name": "",
      "BaseName": "alpine:latest",
      "SourceCode": "FROM alpine:latest",
      "Platform": "",
      "From": {
        "Image": "alpine:latest"
      },
      "Commands": [
        {
          "Maintainer": "Thierry VOYAT \u003cthierry.voyat@ac-amiens.fr\u003e",
          "Name": "maintainer"
        },
        {
          "CmdLine": [
            "apk update     \u0026\u0026 apk add tzdata \u0026\u0026 cp /usr/share/zoneinfo/Europe/Paris /etc/localtime     \u0026\u0026 echo \"Europe/Paris\" \u003e  /etc/timezone     \u0026\u0026 apk del tzdata     \u0026\u0026 apk add         ca-certificates         openssl         nmap         nmap-ncat         nmap-nping         nmap-nselibs         nmap-scripts     \u0026\u0026 update-ca-certificates     \u0026\u0026 rm -rf /var/cache/apk/*"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "Chown": "",
          "From": "",
          "Name": "copy",
          "SourcesAndDest": [
            "FILES/.",
            "/usr/share/nmap/scripts/"
          ]
        },
        {
          "CmdLine": [
            "/usr/bin/nmap"
          ],
          "Name": "entrypoint",
          "PrependShell": false
        },
        {
          "CmdLine": [
            "--help"
          ],
          "Name": "cmd",
          "PrependShell": false
        }
      ]
    }
  ]
}