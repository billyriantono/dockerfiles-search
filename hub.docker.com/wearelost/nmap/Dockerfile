FROM ubuntu

RUN apt-get update && apt-get install -y wget gcc g++ libssl-dev make

WORKDIR /root/nmap-build
WORKDIR /root

RUN wget https://nmap.org/dist/nmap-7.70.tar.bz2 && tar jxf nmap-7.70.tar.bz2

WORKDIR /root/nmap-7.70

RUN ./configure --prefix=/ --with-liblua=included && make && make DESTDIR=/root/nmap-build install

WORKDIR /root/nmap-build/bin

RUN strip nmap
RUN strip ncat
RUN strip nping

FROM scratch

WORKDIR /bin
WORKDIR /lib
WORKDIR /lib64
WORKDIR /share

WORKDIR /

COPY --from=0 /root/nmap-build/bin/nmap /bin/
COPY --from=0 /root/nmap-build/share/nmap /share/
COPY --from=0 /usr/lib/x86_64-linux-gnu/libcrypto.so* /lib/
COPY --from=0 /usr/lib/x86_64-linux-gnu/libssl.so* /lib/
COPY --from=0 /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /lib/
COPY --from=0 /lib/x86_64-linux-gnu/libc.so.6 /lib/
COPY --from=0 /lib/x86_64-linux-gnu/libdl.so.2 /lib/
COPY --from=0 /lib/x86_64-linux-gnu/libm.so.6 /lib/
COPY --from=0 /lib/x86_64-linux-gnu/libgcc_s.so.1 /lib/
COPY --from=0 /lib/x86_64-linux-gnu/libpthread.so.0 /lib/
COPY --from=0 /lib64/ld-linux-x86-64.so.2 /lib64/

ENTRYPOINT ["/bin/nmap"]
