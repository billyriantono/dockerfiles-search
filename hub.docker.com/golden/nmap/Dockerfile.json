{
  "MetaArgs": null,
  "Stages": [
    {
      "Name": "",
      "BaseName": "fedora:latest",
      "SourceCode": "FROM fedora:latest",
      "Platform": "",
      "From": {
        "Image": "fedora:latest"
      },
      "Commands": [
        {
          "CmdLine": [
            "dnf repolist"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "dnf -y distro-sync"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "dnf -y install subversion"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "Chown": "",
          "Name": "add",
          "SourcesAndDest": [
            "https://nmap.org/dist/nmap-6.49BETA4-1.x86_64.rpm",
            "/tmp/"
          ]
        },
        {
          "CmdLine": [
            "dnf -y install /tmp/nmap-6.49BETA4-1.x86_64.rpm"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "rm /tmp/nmap-6.49BETA4-1.x86_64.rpm"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "dnf autoremove"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "dnf clean all"
          ],
          "Name": "run",
          "PrependShell": true
        },
        {
          "CmdLine": [
            "/usr/bin/nmap"
          ],
          "Name": "entrypoint",
          "PrependShell": false
        },
        {
          "CmdLine": [
            "-V"
          ],
          "Name": "cmd",
          "PrependShell": false
        }
      ]
    }
  ]
}